src/compas_rhino/geometry/brep/brep.py

```diff
@@ -0,0 +1,82 @@
+from compas.geometry import Geometry
+from compas_rhino.conversions import box_to_rhino
+
+from Rhino.Geometry import Brep
+
+from .face import RhinoBRepFace
+from .edge import RhinoBRepEdge
+from .vertex import RhinoBRepVertex
+from .loop import RhinoBRepLoop
+
+
+class RhinoBRep(Geometry):
+    def __init__(self):
+        super(RhinoBRep, self).__init__()
+        self.rhino_brep = None
+
+    @property
+    def data(self):
+        faces = []
+        for face in self.faces:
+            faces.append(face.data)
+        return {"faces": faces}
+
+    @data.setter
+    def data(self, data):
+        faces = []
+        for facedata in data["faces"]:
+            face = RhinoBRepFace.from_data(facedata)
+            faces.append(face)
+        self.rhino_brep = BRep.from_faces(faces).rhino_brep
+
+    @classmethod
+    def from_faces(cls, faces):
+        brep = Brep()
```

Comment:
?

Reply:
To be implemented. 

While we're here, this part will have to be a bit different than `compas_occ`. `BrepEdge`, `BrepFace` and even `BrepVertex` all don't have a public constructor and can only be created via the `Brep` instance (e.g. `Brep().Vertices.Add(Point3d)`). 

Consequently, de-serializing a `compas_rhino.geometry.RhinoBRepVertex` alone does not result in a `Rhino.Geometry.BrepVertex` instance. Therefore, I imagine the de-serialized building blocks will be containing COMPAS types and the actual reconstruction of the native Rhino `Brep` will take place here, in `from_faces`. It can then also "inject" the native types to each of the underlying objects.

Do you see a better way to go about it?

Reply:
no, that makes sense to me. in fact, also in `compas_occ` we should try to funnel everything through the `Brep` class, and expose the other types only through the element accessors (`.vertices`, `.edges`, `.loops`, `.faces`), or for development pruposes.